pkg/workflows/wasm/host/module.go

```diff
@@ -393,39 +429,57 @@ func containsCode(err error, code int) bool {
 	return strings.Contains(err.Error(), fmt.Sprintf("exit status %d", code))
 }
 
-func fetchFn(logger logger.Logger, modCfg *ModuleConfig) func(caller *wasmtime.Caller, respptr int32, resplenptr int32, reqptr int32, reqptrlen int32) int32 {
-	const fetchErrSfx = "error calling fetch"
+func createFetchFn(
+	logger logger.Logger,
+	reader unsafeReaderFunc,
+	writer unsafeWriterFunc,
+	sizeWriter unsafeFixedLengthWriterFunc,
+	modCfg *ModuleConfig, store *store,
+) func(caller *wasmtime.Caller, respptr int32, resplenptr int32, reqptr int32, reqptrlen int32) int32 {
+	logFetchErr := func(err error) { logger.Errorf("error calling fetch: %s", err.Error()) }
```

Comment:
Is this helper just for convenience? Because this approach ends up using the wrong `caller` file/line-number. There are work arounds, but it is generally best to inline these.

Reply:
we used to have everything inline but it was getting a bit messy with `Fetch` & `Emit` functions so we started extracting it. you mean the stacktrace/debugging would have the wrong file/line-number?  was not aware of this, Im only modifying the name here, but will def bring this up